core/dispatcher/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -633,8 +638,33 @@ class ContainerPool(
     }
 }
 
-object ContainerPool {
+object ContainerPool extends Logging {
     def requiredProperties = Map(selfDockerEndpoint -> "localhost") ++ Map(dockerImageTag -> "latest") ++ Map(invokerContainerNetwork -> "bridge") ++ Map(invokerContainerPolicy -> "")
     type RunResult = (Instant, Instant, Option[(Int, String)])
-    val defaultMaxActive = 4
+
+    private val defaultNumCore = 4  // The number of cores if it cannot be determined dynamically with nproc
+    // The shareFactor indicates the number of containers that would share a single core, on average.
+    // cpuShare is a docker option (-c) whereby a container's CPU access is limited.
+    // A value of 1024 is the full share so a strict resource division with a shareFactor of 2 would yield 512.
+    // On an idle/underloaded system, a container will still get to use underutilized CPU shares.
+    val shareFactor = 2
```

Comment:
Yes, I think we could consider making this a deployment parameter that is stored in consul and adjustable so that if an invoker is restarted, it will grab the current shareFactor value.  That would make this value here a backup value in case someone writes a garbage value into consul.